Base/usr/share/man/man1/drain.md

```diff
@@ -0,0 +1,38 @@
+## Name
+
+drain - Print file to stdout, while progressively deleting read segments
+
+## Synopsis
+
+```**sh
+$ drain [file]
+```
+
+## Description
+
+drain is a utility for progressively deleting a file while reading segments from it.
+It is useful for example on low memory storage situations, when completing a task on
+a big file can be done when you actually need to delete it, so this utility reads the
+file and delete it in parts, and when utility reads & delete, the utility streams the
+data to another program.
```

Comment:
This is quite a run-on sentence, let's break it up.

And much of this sentence reads quite awkwardly:
* It is useful for example on low memory - commas around "for example", and should be "in" low memory
* "when completing a task on a big file can be done when you actually need to delete it" - I don't actually know what that's saying
* "this utility reads the file and delete it in parts" - should be "deletes", and maybe use "blocks" instead of "parts" to make it clear this ties to the block-size parameter.
* "and when utility reads & delete" - should be "the utility" and "deletes", but is awkwardly worded in general
* "the utility streams the data to another program." - It just writes to stdout, no? It's up to the caller to decide if it gets directed to another program or file

I'm also thinking that low-disk is a more apt use case than low-memory? Not a strong opinion on that though.

Maybe something like:

> `drain` is a utility for simultaneously reading and deleting a file. It is useful, for example, when working with large files on systems with low disk space. `drain` will read blocks from the provided file, and write each block to the standard output stream before removing that block from the file. The output may be redirected to another file or utility for further processing.